src/RamseyCacheKeyGenerator.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace OAT\Library\HealthCheckCache;
+
+use Ramsey\Uuid\Uuid;
+
+final class RamseyCacheKeyGenerator implements CacheKeyGeneratorInterface
+{
+    private const DEFAULT_PREFIX = 'oat-health-check';
+
+    /** @var string */
+    private $prefix;
+
+    public function __construct(string $prefix = null)
```

Comment:
can you please `__construct(?string $prefix = null)` ?

Reply:
what would be the benefit of accepting exactly `null` value also?

Reply:
- to allow key generation without prefix if needed
- and to comply to QA tools that we may add later (like psalm which is very strict :) )

Reply:
On the first hand, this case also allows generating key without providing any prefix. If we add `?string` means we can either pass `null` or `string` as argument but from the behaviour's viewpoint, it is string which has more meaning, isn't it?
On the second hand, I don't think that modifying domain related behaviour just because of testing purposes is a good thing. 

Don't get me wrong, I'm not intended to split hairs just seeing many times usage of `?string $prefix = null` (null as type declaration and null as default value at the same time) and want to understand its benefit. In my opinion, it's much more strict if an argument explicitly accept only one type of value. Furthermore Clean Code does not really advise passing null value too.

Reply:
using `string $prefix = self::DEFAULT_PREFIX` is ok for me